### PR TITLE
Block: remove reference to deprecated getTimeSeconds()

### PR DESCRIPTION
**Pull request by schildbach**
If approved, I'll merge this to 0.17 as well.

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java [hunk 1]
```diff
@@ -589,7 +589,7 @@ private void connectBlock(final Block block, StoredBlock storedPrev, boolean exp
             throw new VerificationException("Block failed checkpoint lockin at " + (storedPrev.getHeight() + 1));
         if (shouldVerifyTransactions()) {
             for (Transaction tx : block.getTransactions())
-                if (!tx.isFinal(storedPrev.getHeight() + 1, block.getTimeSeconds()))
+                if (!tx.isFinal(storedPrev.getHeight() + 1, block.time().getEpochSecond()))
```
Comment (msgilligan) [inline core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java:592]: Won't this need a rebase given a PR that I just approved.

File: core/src/main/java/org/bitcoinj/core/NetworkParameters.java [hunk 2]
```diff
@@ -524,7 +524,7 @@ public EnumSet<Block.VerifyFlag> getBlockVerificationFlags(final Block block,
     public EnumSet<Script.VerifyFlag> getTransactionVerificationFlags(final Block block,
             final Transaction transaction, final VersionTally tally, final Integer height) {
         final EnumSet<Script.VerifyFlag> verifyFlags = EnumSet.noneOf(Script.VerifyFlag.class);
-        if (block.getTimeSeconds() >= NetworkParameters.BIP16_ENFORCE_TIME)
+        if (block.time().getEpochSecond() >= NetworkParameters.BIP16_ENFORCE_TIME)
```
Comment (msgilligan) [inline core/src/main/java/org/bitcoinj/core/NetworkParameters.java:527]: Shouldn't `BIP16_ENFORCE_TIME` be made available as a time type?

File: core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java [hunk 3]
```diff
@@ -774,9 +774,9 @@ private static long getMedianTimestampOfRecentBlocks(StoredBlock storedBlock,
                                                          BlockStore store) throws BlockStoreException {
         long[] timestamps = new long[11];
```
Comment (msgilligan) [inline core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java:775]: I wonder if this should be converted to an array of `Instant`?

File: core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java [hunk 4]
```diff
@@ -600,7 +600,7 @@ private void connectBlock(final Block block, StoredBlock storedPrev, boolean exp
                         block.getHashAsString(), filteredTxHashList.size(), filteredTxn.size());
                 for (Sha256Hash hash : filteredTxHashList) log.debug("  matched tx {}", hash);
             }
-            if (expensiveChecks && block.getTimeSeconds() <= getMedianTimestampOfRecentBlocks(head, blockStore))
+            if (expensiveChecks && block.time().getEpochSecond() <= getMedianTimestampOfRecentBlocks(head, blockStore))
```
Comment (msgilligan) [inline core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java:603]: Can we call a method that returns `Instant` here?

**Review by msgilligan — CHANGES_REQUESTED** [review]
I think this PR shows there are more things that can or should be converted to `Instant`

File: core/src/main/java/org/bitcoinj/core/NetworkParameters.java (hunk 2 above)
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/core/NetworkParameters.java:527]: We could, and perhaps should. But it's out of scope for this PR.

File: core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java (hunk 3 above)
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java:775]: We could rewrite the entire method at any time, including it's return type. It's private. But again, out of scope for this PR.

File: core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java (hunk 4 above)
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java:603]: See above answer, assuming you are talkion about `getMedianTimestampOfRecentBlocks()`.

File: core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java (hunk 1 above)
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java:592]: Sure, I'll rebase.

File: core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java (hunk 3 above)
Comment (msgilligan) [reply core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java:775]: OK

File: core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java (hunk 4 above)
Comment (msgilligan) [reply core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java:603]: Ok

**Review by msgilligan — APPROVED** [review]

File: core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java (hunk 3 above)
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java:775]: See  #3506

**Comment by schildbach** [comment]
Merged to master and 0.17.

**Pull request merged**